components/badge.jsx

```diff
@@ -0,0 +1,62 @@
+var React       = require('react'),
+    ImageTag    = require('./imagetag.jsx');
+
+var Badge = React.createClass({
+    propTypes: {
+        title: React.PropTypes.string.isRequired,
+        status: React.PropTypes.string,
+        icon: React.PropTypes.string.isRequired,
+        icon2x: React.PropTypes.string.isRequired
+    },
+    render: function (){
+
+        /**
+         * returning the Status HTML
+         * @param statusVal
+         * @returns ReactElement
+         */
+        function status (statusVal) {
+            var classNames = 'label label-default ' + statusVal;
+            var statusIcon = "";
+
+            if (statusVal === undefined) {
+                return "";
+            }
+
+            switch (statusVal) {
+                case 'pending':
+                    statusIcon = "<i class='fa fa-fw fa-clock-o'></i>";
+                    break;
+                case 'achieved':
+                    statusIcon = "<i class='fa fa-fw fa-check'></i>";
+                    break;
+                case 'eligible':
+                    statusIcon = "<i class='fa fa-fw fa-pencil'></i>";
+                    break;
+            }
+            return (
+                <div className="status">
+                    <div className={classNames}><span dangerouslySetInnerHTML={{__html: statusIcon}}></span>{statusVal}</div>
+                </div>
+            );
+        }
+
+        var statusElement = "";
```

Comment:
see note above. We can then do this as:
```
var statusElement = this.status();
```
where `this.status()` is responsible for checking `this.props.status`, return `false` if not set, and calling a `.toLowerCase()` when it is.